src/credentials/PersistentStorageOpCertStore.cpp

```diff
@@ -319,6 +386,97 @@ CHIP_ERROR PersistentStorageOpCertStore::UpdateOpCertsForFabric(FabricIndex fabr
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR PersistentStorageOpCertStore::BasicVidVerificationAssumptionsAreMet(FabricIndex fabricIndex) const
+{
+    VerifyOrReturnError(mStorage != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    VerifyOrReturnError(IsValidFabricIndex(fabricIndex), CHIP_ERROR_INVALID_FABRIC_INDEX);
+    // Must already have a valid NOC chain.
+    VerifyOrReturnError(HasNocChainForFabric(fabricIndex), CHIP_ERROR_INCORRECT_STATE);
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PersistentStorageOpCertStore::UpdateVidVerificationSignerCertForFabric(FabricIndex fabricIndex, ByteSpan vvsc)
+{
+    ReturnErrorOnFailure(BasicVidVerificationAssumptionsAreMet(fabricIndex));
+    VerifyOrReturnError(vvsc.empty() || vvsc.size() <= Credentials::kMaxCHIPCertLength, CHIP_ERROR_INVALID_ARGUMENT);
+
+    // Can't try to set a VVSC if ICAC present.
```

Comment:
The API did not document that this is the responsibility of the callee to check.

In general, it would be nice if the cluster implementation could do these checks instead of every opcert store backend having to do them...  Not sure how viable that is.

Reply:
Will do